treeherder/log_parser/tasks.py

```diff
@@ -19,85 +19,82 @@ def parser_task(f):
     """Decorator that ensures that log parsing task has not already run,
     and also adds New Relic annotations.
     """
-    def inner(job_log_id, priority):
-        newrelic.agent.add_custom_parameter("job_log_id", job_log_id)
-        job_log = JobLog.objects.select_related("job").get(id=job_log_id)
+    def inner(job_log):
+        logger.debug("parser_task for %s" % job_log.id)
+        newrelic.agent.add_custom_parameter("job_log_id", job_log.id)
         newrelic.agent.add_custom_parameter("job_log_name", job_log.name)
         newrelic.agent.add_custom_parameter("job_log_url", job_log.url)
         newrelic.agent.add_custom_parameter("job_log_status_prior",
                                             job_log.get_status_display())
         if job_log.status == JobLog.PARSED:
-            logger.info("log already parsed")
+            logger.info("%s log already parsed" % job_log.id)
             return True
 
-        return f(job_log, priority)
+        return f(job_log)
 
     inner.__name__ = f.__name__
     inner.__doc__ = f.__doc__
 
     return inner
 
 
-def parse_job_log(func_name, routing_key, job_log):
-    """
-    Schedule the log-related tasks to parse an individual log
-    """
-    task_funcs = {
-        "store_failure_lines": store_failure_lines,
-        "parse_log": parse_log,
+@retryable_task(name='log-parser', max_retries=10)
+def parse_logs(job_id, job_log_ids, priority):
```

Comment:
But we're then double-annotating with New Relic?

Perhaps we need to split it up?